text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+# Motivation
+[motivation]: #motivation
+
+It is in the context of failure after failure in the OSS ecosystem related to compromised packages and lack of reliable provenance, as well as the looming changes that will be driven by the recently announced White House cybersecurity strategy, that this RFC is presented. The Rust ecosystem can *and should* lead on these topics given its accelerating adoption in mission critical scenarios, expanding use in linux core, etc, etc. Not only can the community address basic artifact signature operations, but it can also “skate to where the puck is going to be” on the evolving build plant ecosystem. 
+
+On the topic of build system integrity, historically, this operational domain was not delineated as a specific function from generic security systems operations in the context of say, NIST. However, with the new SSDF directive NIST has specifically engaged on this topic. from a standards perspective. It’s not an exaggeration to say that NIST artifacts can be a bit dense, but more interestingly, one of the other early ecosystem events post Solar Winds was the launch of the Supply Chain Levels for Software Artifacts or “SLSA”. [SLSA](https://slsa.dev) is a capabilities framework that specifically and exclusively addresses the topic of build system fidelity and fills a gap in the existing standards. It is relatively lean and approachable and provides organizations with a fast, easy way to assess current capabilities and plan for improvement. SLSA addresses both artifact provenance and build system integrity and also very clearly lays out relevant attack vectors as can be seen below):
+
+
+![SLSA: Supply-chain-threats](https://user-images.githubusercontent.com/20165/174750708-2be483ac-7e41-4bc3-8ee9-440ef33d9423.svg)
+
+In the context of the SLSA framework, adopting Sigstore as proposed in this RFC  will immediately address the B, C, F and H attacks identified above.
+
+## Current state of Cargo/ crates.io tooling
+
+At present, crates.io provides the following information about published crates:
+
+* checksum of crate contents
+* crate author/ownership via GitHub
+
+Our current reality is that:
+
+* The verification of the above information relies on a third party (crates.io) not being compromised.
+* In the event that crates.io is compromised, crates pulled from crates.io cannot be verified.
+* There is no way to establish the identity of the system that built the artifact.
+* There is only the limited identity metadata associated with the GH ID that is the source of the artifact.
```

Comment:
This assumes that all sources on crates.io are on GitHub, which is distinctly false.

Reply:
Right now, GitHub is the source of the *identity* of the *people* who publish all crates, but our intention is to have other ways to log in to crates.io in the future. I agree that there is no current requirement for hosting the source code of any crate on GitHub, but it *is* the only way to log in to crates.io right now.